internal/entitlement/balanceworker/worker.go

```diff
@@ -127,24 +168,204 @@ func (w *Worker) handleSystemEvent(msg *message.Message) ([]*message.Message, er
 	}
 
 	switch ceType {
+	// Entitlement events
 	case entitlement.EntitlementCreatedEvent{}.Spec().Type():
 		event, err := spec.ParseCloudEventFromBytes[entitlement.EntitlementCreatedEvent](msg.Payload)
 		if err != nil {
-			w.opts.Logger.Error("failed to parse entitlement created event", w.messageToLogFields(msg)...)
-			return nil, err
+			return nil, fmt.Errorf("failed to parse entitlement created event: %w", err)
```

Comment:
These should be permanent errors, should not go to DLQ.

Reply:
Given we are moving to messagebus, I wouldn't care about this right now. It should handle this part gracefully.